### DISCUSSION TITLE
Hook for UltimateClans V7

### DISCUSSION THREAD

**Discussion by GaetanoDEV** [GriefPrevention/GriefPrevention#2234] [Ideas]
Hi,
I am using UltimateClans (V7)  as plugin and i'll request some features hooking in GP.
UltimateClans has a particular addon called "Lands" , that create a claimed area maneged by this addon, but there is a problem:

- If we make a claimed Land with UClans,  players can use GP Shovel to make a claim in the uclans' land and vice-versa.

Is possible to use the UClans' APIs to make addons or hooks for plugin

API: https://github.com/UlrichBR/UClansV7-API
UltimateClans: https://polymart.org/resource/ultimate-clans.1162
Lands Addon for UClans: https://polymart.org/resource/peaceful-lands-uclans-addon.2513

**Comment by umqashi** [comment]
UP This would be nice feature

**Comment by bigpresh** [comment]
GP fires various events which other plugins can cancel if needed - UltimateClans should be able to listen for the `ClaimCreatedEvent` and cancel it if it's "in the uclans' land", and the claim will not be created.

(It might also need to listen for other related events - e.g. `ClaimExtendEvent` / `ClaimResizeEvent` so that you can't create a claim outside but near an UltimateClans area then resize it to overlap, etc - have a look at the [GP events](https://github.com/GriefPrevention/GriefPrevention/tree/master/src/main/java/me/ryanhamshire/GriefPrevention/events))
